esphome/components/display/display.cpp

```diff
@@ -608,6 +608,66 @@ bool Display::clamp_y_(int y, int h, int &min_y, int &max_y) {
   return min_y < max_y;
 }
 
+const uint8_t font_r[8] PROGMEM = {0x41, 0x7F, 0x7F, 0x09, 0x19, 0x7F, 0x66, 0x00};  // 'R'
+const uint8_t font_g[8] PROGMEM = {0x1C, 0x3E, 0x63, 0x41, 0x51, 0x73, 0x72, 0x00};  // 'G'
+const uint8_t font_b[8] PROGMEM = {0x41, 0x7F, 0x7F, 0x49, 0x49, 0x7F, 0x36, 0x00};  // 'B'
+
+void Display::test_card() {
+  // end
+  int w = get_width(), h = get_height(), image_w, image_h;
+  if (this->get_display_type() == DISPLAY_TYPE_COLOR) {
+    Color R(255, 0, 0), G(0, 255, 0), B(0, 0, 255);
+    if (w > h) {
+      image_w = w > 330 ? 310 : w - 20;
+      image_h = h > 280 ? 255 : h - 20;
+    } else {
+      image_h = h > 330 ? 310 : h - 20;
+      image_w = w > 280 ? 255 : w - 20;
+    }
```

Comment:
When image_h is > 255 the value passed to esp_scale (which takes 8 bit arguments) is truncated.
```suggestion
    image_w = std::min(w - 20, 310);
    image_h = std::min(h - 20, 255);
```